src/sysdir.c

```diff
@@ -85,8 +85,12 @@ static int git_sysdir_guess_template_dirs(git_buf *out)
 
 typedef int (*git_sysdir_guess_cb)(git_buf *out);
 
-static git_buf git_sysdir__dirs[GIT_SYSDIR__MAX] =
-	{ GIT_BUF_INIT, GIT_BUF_INIT, GIT_BUF_INIT, GIT_BUF_INIT, GIT_BUF_INIT };
+struct git_sysdir__dir {
+	int initialized;
```

Comment:
The rationale is that we may not need to query system directories, especially for bare repository users.  They _must_ configure their system directories (if any) after calling `git_libgit2_init`, but they would want to avoid hitting the disk for unnecessary lookups that they were going to override.